packages/devp2p/src/ExchangeProtocol.ts

```diff
@@ -0,0 +1,23 @@
+import ms from 'ms'
+import { EventEmitter } from 'events'
+import { Peer, DISCONNECT_REASONS } from './rlpx/peer'
+
+export class ExchangeProtocol extends EventEmitter {
+  _version: number
+  _peer: Peer
+  _statusTimeoutId: NodeJS.Timeout
+
+  constructor(version: number, peer: Peer) {
+    super()
+
+    this._version = version
+    this._peer = peer
+    this._statusTimeoutId = setTimeout(() => {
+      this._peer.disconnect(DISCONNECT_REASONS.TIMEOUT)
+    }, ms('5s'))
+  }
+
+  getVersion() {
+    return this._version
+  }
+}
```

Comment:
Don't change this yet until there is some more confirmation from other team members, but I am not so satifsfied with the overall (already existing) directory structure and naming here. So to have this file on the top level is somewhat suboptimal respectively too exposed. And these dedicated directories for `eth`, `les` are also somewhat overblown.

I guess we might be good here to rename since this is all re-exported by the main `index.ts` file? 🤔

My suggestion here would be to follow a bit the naming from similar client directory structures and do:

- `src/eth/index.ts` -> `src/protocol/eth.ts`
- `src/les/index.ts` -> `src/protocol/les.ts`
- `src/ExchangeProtocol.ts` -> `src/protocol/protocol.ts`

And also for the class name: `ExchangeProtocol` -> `Protocol`

What do others think?

Reply:
Also, just noting here before I forget, should likely only be done after a renaming though: this new protocol class should be added to the `exclude` directive from the main `typedoc.ts` file so that it doesn't get included into the code documentation on generation.

Maybe a code doc that this class is only internal and shouldn't be directly used would also not hurt, so that this doesn't sneek into the public API.

Reply:
@holgerd77 Just unsure about where to document that `Protocol` shouldn't be directly used. Should that be done inside `README.md` in the `devp2p` package?